packages/data/addon/mirage/handlers/graphql.js

```diff
@@ -18,42 +74,136 @@ function _getSeedForRequest(table, args, fields) {
   return tableLength + argsLength + fieldsLength;
 }
 
+function _createInterval(filters) {
+  const intervals = filters.map(f => DATE_FILTER_OPS[f.operator](f.values, f.grain));
+  const lowerBound = moment.max(intervals.map(interval => interval[0]).filter(Boolean));
+  const upperBound = moment.min(intervals.map(interval => interval[1]).filter(Boolean));
```

Comment:
Is it not? I believe it works for finding the intersection of multiple intervals. Do you have a counter example that it wouldn't work for?

Reply:
```js
const intervals = [
  [‘2020-01-01’, ‘2020-01-04’],
  [‘2020-01-01’, ‘2020-01-02’],
  [‘2020-01-02’, ‘2020-01-03’],
  [‘2020-01-03’, ‘2020-01-04’],
]

const lowerBound = ‘2020-01-03’
const upperBound = ‘2020-01-02’
```

So it should technically be `2020-01-01/2020-01-04`, but this gives `2020-01-03/2020-01-02`. Although we could just accept this as a limitation. Didn’t look too closely, but does it handle invalid intervals?

Reply:
Why should it be `2020-01-01/2020-01-04`? This is condensing multiple intervals on a single time dimension down to one interval so it should create an interval that's an intersection of those. The returned values should be valid for all the filtered intervals right? 

I do need to handle the case where the lowerInterval is greater than the upperInterval though. Invalid intervals should return an empty array for dates

Reply:
I added logic to handle the case where lowerBound is later than upperBound and for intervals missing either an upper or lower bound. I also added some tests to make sure the mock's handling of those cases is correct.

Reply:
You're right, I was thinking of fili intervals rather than time dimension filters (since we `&&` filters together) so that is the expected behavior. Although I wonder how we would request data over multiple distinct intervals given this format.

Maybe we could have 
```js
{
  type: 'timeDimension',
  field: 'dateTime',
  parameters: { grain: 'day' },
  operator: 'bet',
  values: ['2020-01-01', '2020-01-02', '2020-01-04', '2020-01-05']
}
```
which would turn into `dateTime=2020-01-01/2020-01-02,2020-01-04/2020-01-05` for fili or just opt for multiple requests